pyproject.toml

```diff
@@ -68,11 +68,35 @@ archs = [
 ] # Forces arm64 build on x86_64 runner using cross-compilation.
 before-build = "export HOMEBREW_AUTO_UPDATING=0 && brew update && brew install ninja"
 
-[tool.pydocstyle]
-convention = "google"
-
 [tool.mypy]
 ignore_missing_imports = true
 
-[tool.isort]
-line_length = 120
+[tool.black]
+line-length = 120
+
+[tool.ruff]
+line-length = 120
+indent-width = 4
+
+[tool.ruff.lint]
+select = [
+    # "B",
+    # "D", # pydocstyle
+    "E",   # pycodestyle
+    "W",   # pycodestyle
+    "F",   # pyflakes
+    "I",   # isort
+    "RUF", # Ruff-specific rules
+    "UP",  # pyupgrade
+    # "ANN",
+    "ICN",
+    # "SIM",
+    # "PL",
+]
+ignore = []
+
```

Comment:
Maybe add
```
[tool.ruff.pycodestyle]
max-doc-length = 120
```
too?

Seems to be off by default
https://docs.astral.sh/ruff/settings/#pycodestyle-max-doc-length